### PR TITLE
fix gcs config

### LINKED ISSUE #828: gcr storage not work
What can we help you?

### PR DESCRIPTION
**Pull request by soulseen**
fix #828 

Signed-off-by: soulseen <zhuxiaoyang1996@gmail.com>

### CODE REVIEW

File: config/config/assets/registry-config.yaml.tmpl [hunk 1]
```diff
@@ -148,7 +148,7 @@ storage:
   {{- if .Spec.Storage.Driver.Gcs }}
   gcs:
     bucket: {{ .Spec.Storage.Driver.Gcs.Bucket | quote }}
-    keyfile: {{ .Spec.Storage.Driver.Gcs.Container | quote }}
+    keyfile: {{ .Spec.Storage.Driver.Gcs.KeyDataRef | quote }}
```
Comment (bitsf) [inline config/config/assets/registry-config.yaml.tmpl:151]: should point to the file

File: config/config/assets/registry-config.yaml.tmpl [hunk 2]
```diff
@@ -148,7 +148,7 @@ storage:
...
     rootdirectory: {{ .Spec.Storage.Driver.Gcs.PathPrefix | quote }}
     chunksize: {{ .Spec.Storage.Driver.Gcs.ChunkSize | quote }}
```
Comment (bitsf) [inline config/config/assets/registry-config.yaml.tmpl:153]: add `{{- if .Spec.Storage.Driver.Gcs.ChunkSize }}`

File: controllers/goharbor/registry/deployments.go [hunk 3]
```diff
@@ -192,12 +192,12 @@ func (r *Reconciler) GetDeployment(ctx context.Context, registry *goharborv1.Reg
...
 	}
 
-	if registry.Spec.Storage.Driver.Gcs != nil && registry.Spec.Storage.Driver.Gcs.KetDataRef != "" {
+	if registry.Spec.Storage.Driver.Gcs != nil && registry.Spec.Storage.Driver.Gcs.KeyDataRef != "" {
 		volumes = append(volumes, corev1.Volume{
 			Name: "gcs-key",
 			VolumeSource: corev1.VolumeSource{
 				Secret: &corev1.SecretVolumeSource{
-					SecretName: registry.Spec.Storage.Driver.Gcs.KetDataRef,
+					SecretName: registry.Spec.Storage.Driver.Gcs.KeyDataRef,
```
Comment (bitsf) [inline controllers/goharbor/registry/deployments.go:200]: `GcsJSONKeyFilePath` is not correct, should not put in folder `/etc/registry`

File: controllers/goharbor/registry/deployments.go [hunk 4]
```diff
@@ -542,7 +542,7 @@ func (r *Reconciler) ApplyGcsStorageEnvs(ctx context.Context, registry *goharbor
 			SecretKeyRef: &corev1.SecretKeySelector{
 				Key: harbormetav1.SharedSecretKey,
```
Comment (bitsf) [inline controllers/goharbor/registry/deployments.go:543]: Key should also be GCS_KEY_DATA, or use `SharedSecretKey` all the time

File: controllers/goharbor/registry/deployments.go (hunk 3 above)
Comment (soulseen) [reply controllers/goharbor/registry/deployments.go:200]: but according to the definition of helm, the files are placed in the `/etc/registry` directory
https://github.com/goharbor/harbor-helm/blob/master/templates/registry/registry-dpl.yaml#L114

**Review by bitsf — CHANGES_REQUESTED** [review]
chartmuseum pod can't start up
image

**Comment by bitsf** [comment]
test step
1. add a secret with google service token with storage admin role
```
apiVersion: v1
kind: Secret
metadata:
  name: gcs-secretkey
  namespace: cluster-sample-ns
data:
  GCS_KEY_DATA: <base64 encoded of the token file content>
```

2. change storage spec of HarborCluster
```
spec:
  storage:
    kind: Gcs
    spec:
      redirect:
        enable: true
      gcs:
        keyDataRef: gcs-secretkey
        bucket: thebucketname
        pathPrefix: harbor-operator/test
```

**Comment by soulseen** [comment]
@bitsf fixed, pls review again. thanks

**Review by bitsf — APPROVED** [review]

**Review by yxxhero — APPROVED** [review]

**Pull request merged**